### DISCUSSION TITLE
How to achieve horizontal scaling with multiple Nats Server?

### DISCUSSION THREAD

**Discussion by nileshsoni97** [nats-io/nats-server#3198] [Q&A] [answered]
Hello guys i'm thinking to move my production from socket.io to Nats.io and the first question come to my mind is how i can scale it with load balancing and horizontal scaling where each nats server communicate to master or main server please explain it with details some sample code to achieve this.

**Comment by derekcollison** [comment]
If you utilize the client and NGS, that is all included and automatic. We resolve connect.ngs.global to the nearest server with the same cloud affinity.

If you want to do this yourself, we do not necessarily recommend LBs, but folks do try to use them. You could simply have a DNS c-name record with all the A records for the nats-servers in the cluster.

NGS simplest imo..

**Reply by nileshsoni97** [reply]
Don't understand it i repeat my question with more details.
suppose i have a mobile application using nats web-sockets and user base is around 100k and simultaneously connected are around 1000 clients and i'm sending some kind of data around 10kb in every 500 millisecond to active clients.
My machine config are 8gb/4core/160gb-mem.
Now i have to config many nats server horizontally how can i achieve that.

**Comment by derekcollison** [comment, marked as answer]
If they are co-located geo wise, run a 3 node cluster, put all three servers A records in a DNS cname which is what your web clients will use to connect.

**Comment by derekcollison** [comment]
Might recommend 8 core.. But you can easily test to make sure.

**Reply by nileshsoni97** [reply]
Okay thanks for your response do you help me to decide do i need to move my production mobile app which is on socket.io to nats does it improve connection, does it's fast from socket.io

**Comment by derekcollison** [comment]
NATS is usually considered extremely performant. If you have the code to test your app with the NATS client, you could point them at our demo server. wss://demo.nats.io:8443
